### PR TITLE
src: gpu: intel: jit: gemm: improve second-token gemm performance on xe2

### PR DESCRIPTION
**Pull request by hidefromkgb**
# Description

This tackles the 'pathological' INT8 GEMM cases on BMG that used to be 10x slower than their FP16 counterparts (see MFDNN-12419).

Newest second-token BMG perf results:
|   | FP16, ms | INT8+ZP, ms | perf, % |
|---|---|---|---|
|BMG|4.57967657|3.71244054|123.36|

### CODE REVIEW

**Comment by hidefromkgb** [comment]
make test
disable benchdnn_all
disable benchdnn_cpu
enable benchdnn_nightly
enable benchdnn_matmul

**Comment by hidefromkgb** [comment]
make test perf-gpu
set primitive=matmul

File: src/gpu/intel/jit/gemm/selector/db/kernel.db [hunk 1]
```diff
@@ -1200,5 +1200,7 @@ auto _CATALOG_ = kcatalog::toFlatCatalog({
...
+{{'G', "gemm", {"[FO]", "O", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, -1, -1}, {-1, 8, -1}, {-1, -1, -1}, {-1, 8, -1}, {16, 16, 1}, "IAB"}, "at32 am128 aB wg 2x1x8 ikr xaf st acb li ar sb128 bk0 bm0 nmk sys", {16, (LoopType) 255, 128, {(LoopType) 161, (LoopType) 255, (LoopType) 2}, {16777216, 131072, 16777216}, {262144, 131072, 16777216}, {16, 8, 128}, {2, 1, 8}, 1, (WGType) 0, 4357, 0, 1024, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
+{{'G', "gemm", {"[FO]", "O", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 25, -1}, {-1, 32, -1}, {-1, 25, -1}, {-1, 32, -1}, {16, 16, 1}, "IAB"}, "at32+m128@80 am32+m128@80 aB wg 8x1x4 ikr wx2 xaf acb hi pt ar sb128 bk0 sm sn bm0 nmk sys", {16, (LoopType) 255, 128, {(LoopType) 209, (LoopType) 255, (LoopType) 2}, {16777216, 262144, 16777216}, {262144, 262144, 16777216}, {16, 16, 128}, {8, 1, 4}, 2, (WGType) 1, 4357, 0, 8192, {16, 16, 4}, {true, true, true}}, {'W', 1, {256}}}
```
Comment (petercad) [inline src/gpu/intel/jit/gemm/selector/db/kernel.db:1205]: Is this entry a duplicate of line 1203? I didn't spot the difference, except that this one is listed for 25 <= n <= 32, while the other covers 9 <= n <= 32.

**Review by petercad — APPROVED** [review]

**Review by kealan-barbieri — APPROVED** [review]

File: src/gpu/intel/jit/gemm/selector/db/kernel.db (hunk 1 above)
Comment (petercad) [reply src/gpu/intel/jit/gemm/selector/db/kernel.db:1205]: OK, so when A/B sums are needed you want to use this strategy only for n >= 25?

**Pull request merged**